packages/manager/src/features/Volumes/VolumeDrawer/LinodeVolumeCreateForm.tsx

```diff
@@ -0,0 +1,217 @@
+  label: '',
+  linode_id: -1,
+  region: 'none',
+  size: 20,
+  tags: [],
+};
+
+export const LinodeVolumeCreateForm = (props: Props) => {
+  const { linode, onClose, openDetails } = props;
+  const { enqueueSnackbar } = useSnackbar();
+
+  const { data: profile } = useProfile();
+  const { data: grants } = useGrants();
+  const { mutateAsync: createVolume } = useCreateVolumeMutation();
+
+  const disabled = profile?.restricted && !grants?.global.add_volumes;
+
+  const {
+    errors,
+    handleBlur,
+    handleChange,
+    handleSubmit,
+    isSubmitting,
+    resetForm,
+    setFieldValue,
+    status: error,
+    touched,
+    values,
+  } = useFormik({
+    initialValues,
+    async onSubmit(values, { setErrors, setStatus }) {
+      const { config_id, label, size, tags } = values;
+
+      /** Status holds our a general error message */
+      setStatus(undefined);
+
+      try {
+        const volume = await createVolume({
+          config_id:
+            // If the config_id still set to default value of -1, set this to undefined, so volume gets created on back-end according to the API logic
+            config_id === -1 ? undefined : maybeCastToNumber(config_id),
+          label,
+          linode_id: maybeCastToNumber(linode.id),
+          size: maybeCastToNumber(size),
+          tags,
+        });
+        resetEventsPolling();
+        enqueueSnackbar(`Volume scheduled for creation.`, {
+          variant: 'success',
+        });
+        onClose();
+        openDetails(volume);
+        // Analytics Event
+        sendCreateVolumeEvent(`Size: ${size}GB`, origin);
+      } catch (error) {
+        handleFieldErrors(setErrors, error);
+        handleGeneralErrors(
+          setStatus,
+          error,
+          `Unable to create a volume at this time. Please try again later.`
+        );
+      }
+    },
+    validationSchema: CreateVolumeSchema,
+  });
+
+  return (
+    <form onSubmit={handleSubmit}>
+      {disabled && (
+        <Notice
+          text={
+            "You don't have permissions to create a new Volume. Please contact an account administrator for details."
+          }
+          important
+        />
+      )}
+      {error && <Notice text={error} variant="error" />}
+      <Typography
+        sx={(theme) => ({
+          marginBottom: theme.spacing(1.25),
+        })}
+        data-qa-volume-attach-help
+        style={{ marginTop: 24 }}
+        variant="body1"
+      >
+        {`This volume will be immediately scheduled for attachment to ${linode.label} and available to other Linodes in the ${linode.region} data-center.`}
+      </Typography>
+      <Typography
+        sx={(theme) => ({
+          marginBottom: theme.spacing(1.25),
+        })}
+        data-qa-volume-size-help
+        variant="body1"
+      >
+        A single Volume can range from 10 to {MAX_VOLUME_SIZE} gigabytes in size
+        and costs $0.10/GB per month. Up to eight volumes can be attached to a
+        single Linode.
```

Comment:
The existence of this drawer was an oversight on my part with DC pricing. 😅 

I'm a little nervous about including a bug fix for DC pricing in among this larger PR because if anything were to need to get reverted, it'd be kind of a pain to make sure we keep those changes. We also need the following to get $0.10/GB per month out of here. 

I think it would make sense to isolate the DC-pricing related fixes to its own ticket and PR. Any other thoughts?

![Screenshot 2023-09-14 at 12 56 09 PM](https://github.com/linode/manager/assets/114685994/51fa6210-8267-4eb3-a64e-510c1543f893)

```suggestion
            {dcSpecificPricing ? (
              <span>
                A single Volume can range from 10 to 10240 GB in size. Up to
                eight Volumes can be attached to a single Linode. 
              </span>
            ) : (
              <span>
                A single Volume can range from 10 to {MAX_VOLUME_SIZE} GB in
                size and costs $0.10/GB per month. <br />
                Up to eight volumes can be attached to a single Linode.
              </span>
```

Reply:
See https://github.com/linode/manager/pull/9683. 👀